miden-lib/asm/kernels/transaction/api.masm

```diff
@@ -863,20 +953,103 @@ export.get_note_serial_number
 end
 
 #! Tells the transaction kernel that we are about to execute a procedure on a foreign account.
-#!
-#! Inputs:  [KERNEL_PROCEDURE_HASH, foreign_account_id, pad(11)]
-#! Outputs: [pad(16)]
+#! Loads the data of the specified foreign account to the memory and moves the pointer of the 
+#! currently accessing account to it. 
+#!
+#! Inputs:
+#!   Operand stack: [KERNEL_PROCEDURE_HASH, foreign_account_id, pad(11)]
+#!   Advice map: {
+#!     FOREIGN_ACCOUNT_ID: [[foreign_account_id, 0, 0, account_nonce], VAULT_ROOT, STORAGE_ROOT, CODE_ROOT],
+#!     STORAGE_ROOT: [[STORAGE_SLOT_DATA]],
+#!     CODE_ROOT: [[ACCOUNT_PROCEDURE_DATA]]
```

Comment:
Shouldn't this be `CODE_COMMITMENT: [num_procs, [ACCOUNT_PROCEDURE_DATA]]`? At least that's how it is in the `save_account_procedure_data`.

Reply:
Right, I forgot to roll it back. I was trying to do the same trick with the procedure data as we did it with the storage data, where we obtain the number of the slots from the `adv.push_mapvaln` instruction, but it failed to work like that (honestly I already forgot why). I will investigate this problem and try again, as you mentioned in your [comment](https://github.com/0xPolygonMiden/miden-base/pull/896#discussion_r1797501135), but for now I'll just roll back the doc comment.